src/renderer/state/upload/selectors.ts

```diff
@@ -71,169 +66,49 @@ export const getCanUndoUpload = createSelector(
-
-        if (!fileMetadata) {
-          return [...channelRows, ...subImageRows];
-        }
-
-        const allChannelIds = uniq(
-          allMetadataForFile
-            .filter((m) => !!m.channelId)
-            .map((m) => m.channelId) as string[]
-        );
-        const allPositionIndexes: number[] = uniq(
-          allMetadataForFile
-            .filter((m) => !isNil(m.positionIndex))
-            .map((m) => m.positionIndex) as number[]
-        );
-        const allScenes: number[] = uniq(
-          allMetadataForFile
-            .filter((m) => !isNil(m.scene))
-            .map((m) => m.scene) as number[]
-        );
-        const allSubImageNames: string[] = uniq(
-          allMetadataForFile
-            .filter((m) => !isNil(m.subImageName))
-            .map((m) => m.subImageName) as string[]
-        );
-        return [
-          convertToUploadJobRow(
-            fileMetadata,
-            [...channelRows, ...subImageRows],
-            allChannelIds,
-            allPositionIndexes,
-            allScenes,
-            allSubImageNames
-          ),
-        ];
-      }
-    );
-  }
+  [getUpload],
+  (upload): FileModel[] =>
+    Object.values(upload).map((fileMetadata) => ({
+      ...fileMetadata,
+      [AnnotationName.IMAGING_SESSION]:
+        fileMetadata[AnnotationName.IMAGING_SESSION] || [],
+      [AnnotationName.NOTES]: fileMetadata[AnnotationName.NOTES] || [],
+      [AnnotationName.PLATE_BARCODE]:
+        fileMetadata[AnnotationName.PLATE_BARCODE] || [],
+      [AnnotationName.WELL]: fileMetadata[AnnotationName.WELL] || [],
+    }))
 );
 
 export const getFileToAnnotationHasValueMap = createSelector(
-  [getFileToMetadataMap],
-  (metadataGroupedByFile) => {
-    const result: { [file: string]: { [key: string]: boolean } } = {};
-    forEach(metadataGroupedByFile, (allMetadata, file) => {
-      result[file] = allMetadata.reduce((accum, curr) => {
-        forEach(curr, (value: any, key: string) => {
+  [getUpload],
+  (fileToMetadataMap): { [file: string]: { [annotation: string]: boolean } } =>
+    Object.values(fileToMetadataMap).reduce(
+      (fileToAnnotationHasValueMap, fileMetadata) => {
+        fileToAnnotationHasValueMap[fileMetadata.file] = Object.entries(
+          fileMetadata
+        ).reduce((annotationHasValueMap, [annotation, value]) => {
           const currentValueIsEmpty = isArray(value)
             ? isEmpty(value)
             : isNil(value);
-          accum[key] = accum[key] || !currentValueIsEmpty;
-        });
+          annotationHasValueMap[annotation] =
+            annotationHasValueMap[annotation] || !currentValueIsEmpty;
 
-        return accum;
-      }, {} as { [key: string]: boolean });
-    });
-    return result;
-  }
+          return annotationHasValueMap;
+        }, {} as { [annotation: string]: boolean });
+
+        return fileToAnnotationHasValueMap;
+      },
+      {} as { [file: string]: { [annotation: string]: boolean } }
+    )
```

Comment:
This comment is not really about the changes present in this PR, just about how this function could be cleaned up in general. Feel free to ignore and merge.

My understanding of this function is that you are taking a map from filenames to metadata objects, and you are producing a map of filenames to `annotationHasValueMap`s. The structure of an `annotationHasValueMap` is a mapping from annotation names to booleans indicating whether that annotation is present for a particular file.
I think we can greatly simplify this complicated nested-reduce logic, with the following steps. (But I haven't run the code, so perhaps something about this guess is wrong). I'll use the lodash library here, since I see that is already in use in the codebase.
1. On line 97, `fileMetadata` is a regular javascript object whose keys are annotation names. Therefore, the keys are unique and only 98 each `annotation` is only present here once. This means that line 102-103 should be equivalent to `annotationHasValueMap[annotation] = !currentValueIsEmpty;`
2. Following change 1, the inner reduce (line 98) is actually performing the same computation as lodash's `_.mapValues`, and the outer reduce (line 94) is also doing a `_.mapValues`.
3. A `annotationHasValueMap` is really just a Set of annotation names (the annotations which this file has). This could allow us to use something like lodash's `_.pickBy` in place of the inner `_.mapValues`. This third change is independent of the first two and would change the contract of the function, so it may not be worth it.

Applying all three changes, this function might look something like the following:
```
function valueIsEmpty(value) {
    return isArray(value)
            ? isEmpty(value)
            : isNil(value);
)

(fileToMetadataMap): { [file: string]: Set<string> } =>
    _.mapValues(fileToMetadataMap, metadata => new Set(Object.keys(_.pickBy(metadata, valueIsEmpty))));
```

Reply:
Good idea! Applied changes 1 & 2 in this commit d99b06586889a388acb1dfa385beed3b3d93944a. Didn't go forward with 3 since it seems like a neutral change but larger change.